lib/constructs/TemplateConstruct.ts

```diff
@@ -1,61 +1,66 @@
+import * as logs from '@aws-cdk/aws-logs';
+import * as cr from '@aws-cdk/custom-resources';
+import * as lambda from '@aws-cdk/aws-lambda';
 
-export interface TemplateProps { }
+
+export interface TemplateProps { 
+  userArn?: string;
+  vpcId?: string;
+}
 
 export default class Template extends Construct {
   constructor(scope: Construct, id: string, props: TemplateProps) {
     super(scope, id);
     const {  } = props;
 
-    const vpc = Vpc.fromLookup(this, "defaultVpc", {
-      isDefault: true
-    });
-
-    const userArn = process.env.USER_ARN;
+    const vpc = props.vpcId 
+      ? Vpc.fromLookup(this, "userVpc", { vpcId: props.vpcId })
+      : Vpc.fromLookup(this, "defaultVpc", { isDefault: true })
+    ;
 
     const cloudNineInstance = new Ec2Environment(this, "CloudNineEnvironment", {
       vpc
-    })
-
-    const addCloudNineMembership = new AwsCustomResource(this, `addCloudNineMembership`, {
-      installLatestAwsSdk: false,
-      onCreate: {
-        service: "Cloud9",
-        action: "createEnvironmentMembership",
-        parameters: {
-          environmentId: cloudNineInstance.environmentId,
-          permissions: "read-write",
-          userArn
-        },
-        physicalResourceId: PhysicalResourceId.of('id'),
-        // ignoreErrorCodesMatching: ".*",
-      },
-      policy: AwsCustomResourcePolicy.fromSdkCalls({resources: AwsCustomResourcePolicy.ANY_RESOURCE})
     });
-    addCloudNineMembership.node.addDependency(cloudNineInstance);
 
+    // If the user is not same principle as the creator of the CDK stack, this gives us
+    // a way to add additional permission to the Cloud9 instance:
+    if (props.userArn) {
+      const addCloudNineMembership = new AwsCustomResource(this, `addCloudNineMembership`, {
+        installLatestAwsSdk: false,
+        onCreate: {
+          service: "Cloud9",
+          action: "createEnvironmentMembership",
+          parameters: {
+            environmentId: cloudNineInstance.environmentId,
+            permissions: "read-write",
+            userArn: props.userArn
+          },
+          physicalResourceId: PhysicalResourceId.of('id'),
+          // ignoreErrorCodesMatching: ".*",
+        },
+        policy: AwsCustomResourcePolicy.fromSdkCalls({resources: AwsCustomResourcePolicy.ANY_RESOURCE})
+      });
+      addCloudNineMembership.node.addDependency(cloudNineInstance);
+  
+    }
 
-    const dynamoDbTable = new Table(this, "DynamoDBTable", {
-        tableName: `${id}Table`,
+    const dynamoDbTableV1 = new Table(this, "DynamoDBTableV1", {
+        tableName: `${id}-v1`,
         billingMode: BillingMode.PAY_PER_REQUEST,
         removalPolicy: RemovalPolicy.DESTROY,
         partitionKey: {
-          name: "partitionKey",
-          type: AttributeType.STRING
-        },
-        sortKey: {
-          name: "sortKey",
+          name: "id",
```

Comment:
Why did you change this to id? I personally think that the primary keys of the table should have 100% generic names, so you can really use the table flexibly and follow the ol' 1 table design.

Reply:
totally agreed. I wanted to start with a DDB table that follows same mental model as relational, with the idea being that we'll add in additional DDB table(s) that show the evolution to a DDB best practices.